app/models.py

```diff
@@ -1,119 +1,87 @@
+"""
+This files handles all the database logic and instances
+"""
 import itertools
-import jwt
-from datetime import datetime, timedelta
-from itsdangerous import (TimedJSONWebSignatureSerializer
-                          as Serializer, BadSignature, SignatureExpired)
-
 from passlib.apps import custom_app_context as pwd_context
-from flask import abort, current_app
 
-from app import app
+
 class Users:
+    """
+    This class handles all the logic and methods
+    associated with a user
+    """
     user_db = {}
     id_generator = itertools.count(1)
-    def __init__(self, username, password):
-        self.id = next(self.id_generator)
-        self.username = username
+
+    def __init__(self, fname, lname, email, password):
+        self._id = next(self.id_generator)
+        self.fname = fname
+        self.lname = lname
+        self.email = email
         self.password = password
-        
 
     def save(self):
-        # self.user_db[self.id] = {self.username:self.password}
-        self.user_db[self.username] = self.password
+        '''Saves the data to the datastructure dictonary'''
+        self.user_db[self._id] = dict(
+            fname=self.fname,
+            lname=self.lname,
+            email=self.email,
+            password=self.password
+        )
+        self.user_db[self.email] = self.password
         return self
 
-    def check_user(self, username):
+    def check_user(self, email):
         '''
         This method takes in a username and
```

Comment:
Kindly edit your comment to reflect that the function takes in email not username

Reply:
noted